lib/internal/crypto/hash.js

```diff
@@ -140,7 +145,53 @@ Hmac.prototype.digest = function digest(outputEncoding) {
 Hmac.prototype._flush = Hash.prototype._flush;
 Hmac.prototype._transform = Hash.prototype._transform;
 
+function Mac(mac, key, options) {
+  if (!(this instanceof Mac))
+    return new Mac(mac, key, options);
+
+  validateString(mac, 'mac');
+  key = prepareSecretKey(key);
+
+  let nid = EVP_PKEY_HMAC;
+  if (mac === 'poly1305') nid = EVP_PKEY_POLY1305;
+  else if (mac === 'siphash') nid = EVP_PKEY_SIPHASH;
+  else if (mac.startsWith('cmac:')) {
+    nid = EVP_PKEY_CMAC;
+    mac = mac.slice(5);
+  }
```

Comment:
Wouldn't it make sense to treat HMAC just like any other MAC algorithm, instead of making it the default? So the first argument always correlates with the NID, and the hash function needs to be specified as an option, both for HMAC and CMAC?